reference/performances_guide/index.rst

```diff
@@ -40,3 +56,192 @@ Here is the complete formula to check if you have, by far, more indexes than the
+      - the product grid loading due to the synchronous loading of the family filter (Community Edition)
+      - the creation of a product (Community Edition)
+      - the configure step of the product / mass edit / change family (Community Edition)
+      - the creation of a channel due to the creation of families requirements (Community Edition)
+
+    **If you plan to use the PIM with more than 10k families, please contact us.**
+
+More than 10k categories?
+-------------------------
+
+The number of categories will impact performances of the PIM in several ways.
+We've tested performances with a set of 10k categories in total.
+
+.. warning::
+
+    On this axis, we still have different issues. We've tested with 10k categories and a low amount of families and attributes.
+
+    Screens impacted are the following and we're releasing performances fixes as 1.4 patches:
+      - change and save permissions on the root category of a large tree (Enterprise Edition)
+
+    **If you plan to use the PIM with more than 10k categories, please contact us.**
+
+Export more than 100k products?
+-------------------------------
+
+We had use cases where customers would export 270k products and had issues with the memory usage.
+
+Most of the massive operations of the PIM as imports and exports are bulked by page of lines (or objects) to avoid too large memory usage.
+
+As each product can have different properties, the export kept the transformed array in memory to add missing columns from a line to another.
+
+In the version 1.4.9 (PIM-5127), we changed the internal behaviour of the CsvProductWriter to use a file buffer to temporary write each transformed array before to aggregate the final result.
+
+So, in a product export, the limitation is not anymore the memory size but the hard drive space.
+
+Please notice that the amount of product values per product will have an impact on the time of execution and memory usage.
+
+In the future version 1.5, this CsvProductWriter is reworked to extract the Buffer component and allow to use this component in other contexts.
+
+MongoDB storage and bulk product saving performances?
+-----------------------------------------------------
+
+In the version 1.3, we have a default product saver which handles the saving of a single product or a set of products in the database.
+
+This default saver is the same for Doctrine ORM and Doctrine MongoDB as it depends on Doctrine Common ObjectManager interface, it uses the default persist/flush operations.
+
+These operations are especially greedy. In the 1.3, to prevent performance issue, we provided another method for the product import :doc:`/cookbook/import_export/mongodb-fast-writer`
+
+Mainly this new method bypass Doctrine MongoDB by normalizing the product objects to final Mongodb Documents to be able to directly save them in the database. It still uses low-level Doctrine methods. It simply doesn't use the UnitOfWork.
```

Comment:
bypassed ... normalizing into final MongoDB so they could directly be written in the database.